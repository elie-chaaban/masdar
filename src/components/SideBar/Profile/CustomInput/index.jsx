import React from 'react';
import {Field} from 'formik';
import PropTypes from 'prop-types';

const CustomInput = ({className = 'col-6', type = 'text', name, enabled, title = null, error, touched}) => (
  <div className={`mt-2 ${className}`}>
    <p>{title ? title : name}</p>
    <Field
      className={`form-control ${enabled ? 'active' : ''}`}
      type={type}
      placeholder={title ? title : name}
      name={name}
      disabled={!enabled}
    />
    {error && touched ? <div className="text-danger">{error}</div> : null}
  </div>
);

CustomInput.protoTypes = {
  name: PropTypes.string.isRequired,
  enabled: PropTypes.bool.isRequired,
  title: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string,
  error: PropTypes.string,
  touched: PropTypes.bool.isRequired
};

export default CustomInput;
