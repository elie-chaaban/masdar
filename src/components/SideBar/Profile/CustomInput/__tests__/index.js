import React from 'react';
import {shallow} from 'enzyme';
import CustomInput from '..';

test('Custom Input Component', () => {
  let dom = shallow(<CustomInput name="user" enabled={true} touched={false} />);
  expect(dom).toMatchSnapshot();
  expect(dom.find('.text-danger')).toHaveLength(0);
  expect(dom.find('p').text()).toBe('user');
  expect(dom.find('.col-6')).toHaveLength(1);
  expect(dom.find('.active')).toHaveLength(1);
  expect(dom.find('Field').prop('placeholder')).toBe('user');
  expect(dom.find('Field').prop('disabled')).toBeFalsy();
  expect(dom.find('Field').prop('type')).toBe('text');
  dom.setProps({title: 'userName', type: 'email', className: 'col-4', error: 'invalid user name', touched: true, enabled: false});
  expect(dom.find('.col-6')).toHaveLength(0);
  expect(dom.find('.col-4')).toHaveLength(1);
  expect(dom.find('.active')).toHaveLength(0);
  expect(dom.find('Field').prop('disabled')).toBeTruthy();
  expect(dom.find('p').text()).toBe('userName');
  expect(dom.find('Field').prop('placeholder')).toBe('userName');
  expect(dom.find('.text-danger')).toHaveLength(1);
  expect(dom.find('.text-danger').text()).toBe('invalid user name');
});
