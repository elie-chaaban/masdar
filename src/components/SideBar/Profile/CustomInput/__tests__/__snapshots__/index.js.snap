// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Input Component 1`] = `
<div
  className="mt-2 col-6"
>
  <p>
    user
  </p>
  <Field
    className="form-control active"
    disabled={false}
    name="user"
    placeholder="user"
    type="text"
  />
</div>
`;
